shell/platform/darwin/ios/framework/Source/FlutterStandardCodec.mm

```diff
@@ -0,0 +1,411 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "FlutterStandardCodec_Internal.h"
+
+#pragma mark - Codec for basic message channel
+
+@implementation FlutterStandardMessageCodec
++ (instancetype) shared {
+  static id _shared = nil;
+  if (!_shared) {
+     _shared = [FlutterStandardMessageCodec new];
+  }
+  return _shared;
+}
+
+- (NSData*) encode:(id)message {
+  NSMutableData* data = [NSMutableData dataWithCapacity:32];
+  FlutterStandardWriter* writer = [FlutterStandardWriter withData:data];
+  [writer writeValue:message];
+  return data;
+}
+
+- (id) decode:(NSData*)message {
+  FlutterStandardReader* reader = [FlutterStandardReader withData:message];
+  id value = [reader readValue];
+  if ([reader hasMore]) {
+    @throw [NSException exceptionWithName:NSInvalidArgumentException
```

Comment:
These error situations represent programming errors. Do we really want to handle those using `NSError`s?